web-app/js/portal/prototypes/Ext.js

```diff
@@ -34,5 +34,16 @@ Ext.tree.TreeNode.prototype.clone = function(recursive) {
     }
     
     return clone;
-}
+};
 
+Ext.tree.TreeNode.prototype.getHierarchy = function(attribute) {
```

Comment:
I think a subclass of TreeNode would be cleaner from the treenode's point of view  but overriding the loader createNode method to create the subclassed node is not very clean (would need to expose the default treeloader processing).

This is adding an additional 'getter' style method that won't mutate the object so I don't think this is a big issue.